xbmc/interfaces/builtins/PlayerBuiltins.cpp

```diff
@@ -430,19 +435,27 @@ static int PlayMedia(const std::vector<std::string>& params)
       items.Sort(SortByLabel, SortOrderAscending);
 
     int playlist = containsVideo? PLAYLIST_VIDEO : PLAYLIST_MUSIC;;
-    if (containsMusic && containsVideo) //mixed content found in the folder
+    if ((containsMusic && containsVideo) || nowatched) // only needed on mixed content or unwanted watched/played content
```

Comment:
I would use two loops to simplify the whole interference between `nowatched`, `containsMusic` and `containsVideo`. So first remove all watched items and then use the existing loop as a second loop.